### PR TITLE
fix: region logical regions after catching up

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This PR registers metric logical regions after invoking catchup and fixes metric logical regions not found issues after migrating a physical region.

After replaying the WAL of a metric metadata region, there are two places required to register logical regions manually:
- Metric engine
- RegionServer

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by killme2008** [comment]
@WenyXu Please describe the PR in detail. What issue does it address? What happened before this patch etc.

File: src/datanode/src/region_server.rs [hunk 1]
```diff
@@ -548,6 +548,18 @@ impl RegionServerInner {
...
+                    RegionEngineWithStatus::Registering(_) => {
+                        return error::RegionNotReadySnafu { region_id }.fail()
+                    }
+                    RegionEngineWithStatus::Deregistering(_) => {
+                        return error::RegionNotFoundSnafu { region_id }.fail()
+                    }
+                    RegionEngineWithStatus::Ready(engine) => engine,
+                },
+                None => return error::RegionNotFoundSnafu { region_id }.fail(),
+            },
```
Comment (fengjiachun) [inline src/datanode/src/region_server.rs:562]: `RegionChange::Catchup` and `RegionChange::None` are completely same, can we merge them?

Comment (WenyXu) [reply src/datanode/src/region_server.rs:562]: `RegionChange::None` includes query operations; we don't need to register logical regions after a query.

Comment (WenyXu) [reply src/datanode/src/region_server.rs]: Got you, updated

**Review by waynexia — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**